src/Umbraco.Web/Trees/ContentTreeController.cs

```diff
@@ -306,20 +303,25 @@ private void EnsureName(IEntitySlim entity, string culture)
                 entity.Name = "[[" + entity.Id + "]]";
         }
 
-        private void AddActionNode<TAction>(IUmbracoEntity item, MenuItemCollection menu, bool hasSeparator = false, bool convert = false)
+        //fixme: Remove the need for converting to legacy
+        private void AddActionNode<TAction>(IUmbracoEntity item, MenuItemCollection menu, bool hasSeparator = false, bool convert = false, bool opensDialog = false)
             where TAction : IAction
         {
+            //fixme: Inject
             var menuItem = menu.Items.Add<TAction>(Services.TextService.Localize("actions", Current.Actions.GetAction<TAction>().Alias), hasSeparator);
             if (convert) menuItem.ConvertLegacyMenuItem(item, "content", "content");
+            menuItem.OpensDialog = opensDialog;
         }
 
-        private void AddActionNode<TItem, TAction>(IUmbracoEntity item, MenuItemCollection menu, bool hasSeparator = false, bool convert = false)
-            where TItem : MenuItem, new()
-            where TAction : IAction
-        {
-            var menuItem = menu.Items.Add<TItem, TAction>(Services.TextService.Localize("actions", Current.Actions.GetAction<TAction>().Alias), hasSeparator);
-            if (convert) menuItem.ConvertLegacyMenuItem(item, "content", "content");
-        }
+        ////fixme: Remove the need for converting to legacy
```

Comment:
Why not just remove this as opposed to commenting this out, or is this method a reminder to do fixme cleanup?!

Reply:
yup probably just missed this, should be removed